spec/controllers/application_controller/ci_processing_spec.rb

```diff
@@ -120,9 +120,25 @@
                                         :controller             => "ems_cloud"
                                       )
       allow(controller).to receive(:drop_breadcrumb)
+      module ManageIQ::Providers::Example
+        class CloudManager < ManageIQ::Providers::CloudManager
+          def self.supports_discovery?
+            true
+          end
+
+          def self.ems_type
+            'example'
+          end
+
+          def self.description
+            'example Manager'
+          end
+        end
+      end
       controller.send(:discover)
       expect(response.status).to eq(200)
-      expect(controller.instance_variable_get(:@discover_type)).to include(["Azure", "azure"], ["Amazon", "amazon"])
+      expect(controller.instance_variable_get(:@discover_type)).to include(["example Manager", "example"])
+      ManageIQ::Providers.send(:remove_const, 'Example')
```

Comment:
Your way here is the most literal way to do it, really. Else I'd just append the example on to the actual: `expect(Providers::CloudManager).to receive(:subclasses).and_return(children + [ManageIQ::Providers::Example::CloudManager])` (without actually subclassing the example from CloudManager; where `children` are the actual subclasses you assign before this stub)